language/move-prover/boogie-backend/src/spec_translator.rs

```diff
@@ -300,11 +301,153 @@ impl<'env> SpecTranslator<'env> {
     }
 }
 
+// Boxing/Unboxing
```

Comment:
@shazqadeer @matbd New box/unbox optimization. I think we can use this as the basis for getting closer to monomorphic translation. The next step after this PR would be to make all temporaries (move bytecode parameters and spec language locals) which have a concrete type unboxed, but keep them boxed if they have a generic type `T`, then introduce box/unbox when T transitions to/from a concrete type representation.

Reply:
Yes, makes sense!